bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/internal/items/ItemBuilderTest.java

```diff
@@ -0,0 +1,176 @@
+ */
+package org.eclipse.smarthome.core.internal.items;
+
+import static org.junit.Assert.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+import org.eclipse.smarthome.core.items.ActiveItem;
+import org.eclipse.smarthome.core.items.GroupFunction;
+import org.eclipse.smarthome.core.items.GroupItem;
+import org.eclipse.smarthome.core.items.Item;
+import org.eclipse.smarthome.core.items.ItemFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+/**
+ *
+ * @author Simon Kaufmann - initial contribution and API
+ *
+ */
+public class ItemBuilderTest {
+
+    private ItemRegistryImpl itemRegistry;
+    private @Mock ItemFactory mockFactory;
+    private @Mock ActiveItem mockItem;
+    private @Mock Item originalItem;
+
+    @Before
+    public void setup() {
+        initMocks(this);
+        itemRegistry = new ItemRegistryImpl();
+        itemRegistry.addItemFactory(mockFactory);
+    }
+
+    @Test
+    public void testMinimal() {
+        when(mockFactory.createItem(anyString(), anyString())).thenReturn(mockItem);
+
+        Item res = itemRegistry.newItemBuilder("String", "test").build();
+
+        assertSame(mockItem, res);
+        verify(mockFactory).createItem(eq("String"), eq("test"));
+        verify(mockItem).setLabel(isNull());
+        verify(mockItem).setCategory(isNull());
+        verify(mockItem).addGroupNames(eq(Collections.emptyList()));
+    }
+
+    @Test
+    public void testMinimalGroupItem() {
+        Item resItem = itemRegistry.newItemBuilder("Group", "test").build();
+
+        assertEquals(GroupItem.class, resItem.getClass());
+        GroupItem res = (GroupItem) resItem;
+        verifyNoMoreInteractions(mockFactory);
+        assertNull(res.getCategory());
+        assertEquals(Collections.emptyList(), res.getGroupNames());
+        assertNull(res.getLabel());
+        assertNull(res.getFunction());
+        assertNull(res.getBaseItem());
+    }
+
+    @Test
+    public void testFull() {
+        when(mockFactory.createItem(anyString(), anyString())).thenReturn(mockItem);
+
+        Item res = itemRegistry.newItemBuilder("String", "test") //
+                .withCategory("category") //
+                .withGroups("a", "b") //
+                .withLabel("label") //
+                .build();
+
+        assertSame(mockItem, res);
+        verify(mockFactory).createItem(eq("String"), eq("test"));
+        verify(mockItem).setCategory(eq("category"));
+        verify(mockItem).addGroupNames(eq(Arrays.asList("a", "b")));
+        verify(mockItem).setLabel(eq("label"));
+    }
+
+    @Test
+    public void testFullGroupItem() {
+        Item baseItem = mock(Item.class);
+        GroupFunction mockFunction = mock(GroupFunction.class);
+
+        Item resItem = itemRegistry.newItemBuilder("Group", "test") //
```

Comment:
You could use `GroupItem.TYPE` instead of "Group" since tests are usually a good example that people copy.

Reply:
No. I want this test to break if somebody ever had the stupid idea to change the constant's value!